File: JavaScript/7js 1/ems_8.js
/*
|--------------------------------------------------------------------------
| Exclamation marks series #8: Move all exclamation marks to the end of the sentence
|--------------------------------------------------------------------------
|
*/

function remove(s) {
   return s.replace(/!/g, "") + s.replace(/[^!]/g, "");
}


function remove(s) {
   let x = s.split("").filter((el) => el === "!");
   let y = s.split("").filter((el) => el !== "!");
   return y.join("") + x.join("");
}


function remove(s) {
   return s.replace(/\!/g, "") + "!".repeat((s.match(/\!/g) || []).length);
}


console.log(remove("Hi!"), "Hi!");
console.log(remove("Hi! Hi!"), "Hi Hi!!");
console.log(remove("Hi! Hi! Hi!"), "Hi Hi Hi!!!");
console.log(remove("Hi! !Hi Hi!"), "Hi Hi Hi!!!");
console.log(remove("Hi! Hi!! Hi!"), "Hi Hi Hi!!!!");

File: Python/7py 1/ems_8.py
"""
|--------------------------------------------------------------------------
| Exclamation marks series #8: Move all exclamation marks to the end of the sentence
|--------------------------------------------------------------------------
|
"""

from re import sub
def remove(s):
   return s.replace('!', '') + sub('[^!]', '', s)


def remove(s):
   return ''.join(sorted(s, key=lambda a: a == '!'))


def remove(s):
   return s.replace('!','') + s.count('!') * '!'


print(remove("Hi!"), "Hi!")
print(remove("Hi! Hi!"), "Hi Hi!!")
print(remove("Hi! Hi! Hi!"), "Hi Hi Hi!!!")
print(remove("Hi! !Hi Hi!"), "Hi Hi Hi!!!")
print(remove("Hi! Hi!! Hi!"), "Hi Hi Hi!!!!")